util/rate_limiter.cc

```diff
@@ -144,6 +143,12 @@ void GenericRateLimiter::Request(int64_t bytes, const Env::IOPriority pri,
     available_bytes_ -= bytes;
     total_bytes_through_[pri] += bytes;
     return;
+  } else if (available_bytes_ > 0) {
+    // exhaust all available bytes before queuing the remaining
+    bytes -= available_bytes_;
+    total_bytes_through_[pri] += available_bytes_;
+    available_bytes_ = 0;
+    TEST_SYNC_POINT("GenericRateLimiter::Request:NoAvailableBytes");
```

Comment:
Do we still need to if-else condition on available_bytes_ >= bytes? Can we grant available_bytes_ anyway and depending on how many bytes is still needed, either return or enqueue?